src/Nethermind/Nethermind.AuRa/AuRaSealValidator.cs

```diff
@@ -144,35 +144,90 @@ private Address GetSealer(BlockHeader header)
 
         private class ReceivedSteps
         {
-            private readonly List<(long Step, Address Author, ISet<Address> Authors)> _list = new List<(long Step, Address Author, ISet<Address> Authors)>();
+            private struct AuthorBlock : IEquatable<AuthorBlock>
+            {
+                public AuthorBlock(Address author, Keccak block)
+                {
+                    Author = author;
+                    Block = block;
+                }
+
+                public Address Author { get; }
+                public Keccak Block { get; }
+
+                public bool Equals(AuthorBlock other) => Equals(Author, other.Author) && Equals(Block, other.Block);
+                public override bool Equals(object obj) => obj is AuthorBlock other && Equals(other);
+                public override int GetHashCode() => HashCode.Combine(Author, Block);
+                public static bool operator==(AuthorBlock obj1, AuthorBlock obj2) => obj1.Equals(obj2);
+                public static bool operator!=(AuthorBlock obj1, AuthorBlock obj2) => !obj1.Equals(obj2);
+            }
+            
+            private class AuthorBLockForStep
+            {
+                public AuthorBLockForStep(in long step, AuthorBlock? authorBlock)
```

Comment:
can be IComparable?

Reply:
It was kind of non-standard comparison - hence separate comparer.